	<launch>
		<!--ロボ1の名前空間-->>
		<group ns="hsr1">
			<arg name="sub_joint_state_topic_name"        default="/hsr1/hsrb/joint_states" />
			<arg name="pub_base_twist_topic_name"         default="/hsr1/hsrb/command_velocity" />
			<arg name="pub_arm_trajectory_topic_name"     default="/hsr1/hsrb/arm_trajectory_controller/command" />
			<arg name="pub_gripper_trajectory_topic_name" default="/hsr1/hsrb/gripper_controller/command" />

			<node name="human_navigation_sample" pkg="human_navigation" type="human_navigation_sample" output="screen" launch-prefix="xterm -e">
				<param name="sub_joint_state_topic_name"        type="string" value="$(arg sub_joint_state_topic_name)" />
				<param name="pub_base_twist_topic_name"         type="string" value="$(arg pub_base_twist_topic_name)" />
				<param name="pub_arm_trajectory_topic_name"     type="string" value="$(arg pub_arm_trajectory_topic_name)" />
				<param name="pub_gripper_trajectory_topic_name" type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
			</node>

			<node name="human_navigation_hsr_key_teleop" pkg="human_navigation" type="human_navigation_hsr_key_teleop" output="screen" launch-prefix="xterm -e">
				<param name="sub_joint_state_topic_name"        type="string" value="$(arg sub_joint_state_topic_name)" />
				<param name="pub_base_twist_topic_name"         type="string" value="$(arg pub_base_twist_topic_name)" />
				<param name="pub_arm_trajectory_topic_name"     type="string" value="$(arg pub_arm_trajectory_topic_name)" />
				<param name="pub_gripper_trajectory_topic_name" type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
			</node>
		</group>

		<!--ロボ2の名前空間-->>
		<group ns="hsr2">
			<arg name="sub_joint_state_topic_name"        default="/hsr2/hsrb/joint_states" />
			<arg name="pub_base_twist_topic_name"         default="/hsr2/hsrb/command_velocity" />
			<arg name="pub_arm_trajectory_topic_name"     default="/hsr2/hsrb/arm_trajectory_controller/command" />
			<arg name="pub_gripper_trajectory_topic_name" default="/hsr2/hsrb/gripper_controller/command" />

			<node name="human_navigation_sample" pkg="human_navigation" type="human_navigation_sample" output="screen" launch-prefix="xterm -e">
				<param name="sub_joint_state_topic_name"        type="string" value="$(arg sub_joint_state_topic_name)" />
				<param name="pub_base_twist_topic_name"         type="string" value="$(arg pub_base_twist_topic_name)" />
				<param name="pub_arm_trajectory_topic_name"     type="string" value="$(arg pub_arm_trajectory_topic_name)" />
				<param name="pub_gripper_trajectory_topic_name" type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
			</node>

			<node name="human_navigation_hsr_key_teleop" pkg="human_navigation" type="human_navigation_hsr_key_teleop" output="screen" launch-prefix="xterm -e">
				<param name="sub_joint_state_topic_name"        type="string" value="$(arg sub_joint_state_topic_name)" />
				<param name="pub_base_twist_topic_name"         type="string" value="$(arg pub_base_twist_topic_name)" />
				<param name="pub_arm_trajectory_topic_name"     type="string" value="$(arg pub_arm_trajectory_topic_name)" />
				<param name="pub_gripper_trajectory_topic_name" type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
			</node>
		</group>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port) $(arg sync_time_num)"/>
	</group>

	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" >
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find human_navigation)/launch/hsr.rviz"/>

</launch>
